azure-functions-powershell-library/BindingExtractor.cs

```diff
@@ -0,0 +1,124 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using AzureFunctions.PowerShell.SDK.BundledBindings;
+using Common;
+using System.Management.Automation.Language;
+using System.Reflection;
+
+namespace AzureFunctions.PowerShell.SDK
+{
+    internal static class BindingExtractor
+    {
+        static BindingExtractor()
+        {
+            // We want this behavior to be extensible by third parties who may want to add binding definitions. 
+            // To accomplish this, we parse through all loaded assemblies to find any class which implements 
+            //   IBinding, and load it manually into our supportedBindings
+
+            // Find all types that implement IBinding
+            // Apparently, there are scenarios where this throws a ReflectionTypeLoadException - filtering it seems to help
```

Comment:
The ReflectionTypeLoadException seemed only to happen in really specific cases in the virtualized environments used by the GitHub build/test action. I think it may have had to do with dependency conflicts? Anyway, this comment is a carryover from a previous attempt at a solution that didn't work. I'll improve the documentation around the try/catch block here. 

As for extensibility, TBH, I didn't initially write it this way with extensibility in mind. I was originally going to register the binding classes in their own static constructors, but it turns out that .NET does not run static constructors until a method from the class is called. So, I ended up writing code to instantiate all the classes that implement IBinding, and that evolved to the state that it is in now. 

The benefit of this is the idea is that anyone can implement IInputBinding or IOutputBinding in any assembly, and as long as that assembly is loaded at the time the static constructor for BindingExtractor runs, those classes will be made available. As for how third-party assemblies will get access to those interfaces - I'm assuming we will want to publish this library as a NuGet package, should we ever find a need for someone to write their own bindings.